freud/locality.pyx

```diff
@@ -201,7 +201,7 @@ cdef class NeighborQueryResult:
             yield (npoint.query_point_idx, npoint.point_idx, npoint.distance)
             npoint = dereference(iterator).next()
 
-        raise StopIteration
+        return
```

Comment:
Is this necessary at all? Can you just remove this `return`? I think Python generators raise `StopIteration` automatically, we probably don't need to do anything.
```suggestion
```